siliconcompiler/apps/sc_remote.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2023 Silicon Compiler Authors. All Rights Reserved.
+import argparse
+import json
+import os
+import sys
+import time
+
+from siliconcompiler import Chip
+from siliconcompiler.remote.client import (cancel_job, check_progress, delete_job,
+                                           remote_ping, remote_run_loop)
+from siliconcompiler.utils import default_credentials_file
+
+
+def main():
+    progname = "sc-remote"
+    description = """
+    -----------------------------------------------------------
+    SC app that provides an entry point to common remote / server
+    interactions. Can be used to:
+    * Check software versions on the server (no flags)
+    * Check an ongoing job's progress (-jobid)
+    * Cancel an ongoing job (-jobid + -cancel)
+    * Re-attach SC client to an ongoing job (-jobid + -attach)
+    -----------------------------------------------------------
+    """
+
+    # Argument Parser
+    parser = argparse.ArgumentParser(prog=progname,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter,
+                                     description=description)
+    parser.add_argument('-jobid', required=False)
+    parser.add_argument('-cfg', required=False)
+    parser.add_argument('-reconnect', action='store_true', required=False)
+    parser.add_argument('-cancel', action='store_true', required=False)
+    parser.add_argument('-delete', action='store_true', required=False)
+    parser.add_argument('-credentials', nargs='?', default=default_credentials_file())
```

Comment:
I think we should use the SC command line call so it can handle the loading of the CFG and credentials. also nice to have the same banner printed for the sc-apps

Reply:
Can we add extra parameters with `chip.create_cmdline`? The schema doesn't have equivalents of `-reconnect`, `-cancel`, `-delete`.

Reply:
yes, https://docs.siliconcompiler.com/en/latest/reference_manual/core_api.html#siliconcompiler.core.Chip.create_cmdline